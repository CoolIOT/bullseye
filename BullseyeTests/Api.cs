#if NETCOREAPP2_0
namespace BullseyeTests
{
    using System;
    using System.Runtime.CompilerServices;
    using System.Runtime.InteropServices;
    using ApprovalTests;
    using ApprovalTests.Namers.StackTraceParsers;
    using ApprovalTests.Reporters;
    using ApprovalTests.StackTraceParsers;
    using Bullseye;
    using PublicApiGenerator;
    using Xunit;
    using Xunit.Sdk;

    public class Api
    {
        static Api() => StackTraceParser.AddParser(new WindowsFactStackTraceParser());

        [WindowsFact]
        [UseReporter(typeof(QuietReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void IsUnchanged() =>
            Approvals.Verify(ApiGenerator.GeneratePublicApi(typeof(Targets).Assembly, new[] { typeof(Targets) }));

        [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
        [XunitTestCaseDiscoverer("Xunit.Sdk.FactDiscoverer", "xunit.execution.{Platform}")]
        class WindowsFactAttribute : FactAttribute
        {
            public WindowsFactAttribute() : base() =>
                this.Skip = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? null : "Only works on Windows.";
        }

        class WindowsFactStackTraceParser : XUnitStackTraceParser
        {
            protected override string GetAttributeType() => "BullseyeTests.Api+WindowsFactAttribute";
        }
    }
}
#endif
