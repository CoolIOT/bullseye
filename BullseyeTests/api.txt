[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace Bullseye
{
    public class InvalidUsageException : System.Exception
    {
        public InvalidUsageException() { }
        public InvalidUsageException(string message) { }
        public InvalidUsageException(string message, System.Exception innerException) { }
    }
    public class TargetFailedException : System.Exception
    {
        public TargetFailedException() { }
        public TargetFailedException(string message) { }
        public TargetFailedException(string message, System.Exception innerException) { }
    }
    public class Targets
    {
        public Targets() { }
        public void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        public void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        public void Add<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public void Add(string name, System.Func<System.Threading.Tasks.Task> action) { }
        public void Add(string name, System.Action action) { }
        public void Add<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public void Add<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        public void Add<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static string[] DependsOn(params string[] dependencies) { }
        public static TInput[] ForEach<TInput>(params TInput[] inputs) { }
        public void RunAndExit(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public System.Threading.Tasks.Task RunAndExitAsync(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public static void RunTargetsAndExit(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public static System.Threading.Tasks.Task RunTargetsAndExitAsync(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public static void RunTargetsWithoutExiting(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public static System.Threading.Tasks.Task RunTargetsWithoutExitingAsync(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public void RunWithoutExiting(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public System.Threading.Tasks.Task RunWithoutExitingAsync(System.Collections.Generic.IEnumerable<string> args, System.Func<System.Exception, bool> messageOnly = null, string logPrefix = null) { }
        public static void Target(string name, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Action action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
    }
}