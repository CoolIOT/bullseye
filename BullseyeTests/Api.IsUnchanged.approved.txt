[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace Bullseye
{
    public class static Targets
    {
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0. Will be removed in versio" +
            "n 2.0.0.")]
        public static void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Func<System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Action action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        [System.ObsoleteAttribute("Use Target() instead. Will be removed in version 2.0.0.")]
        public static void Add<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static string[] DependsOn(params string[] dependencies) { }
        public static TInput[] ForEach<TInput>(params TInput[] inputs) { }
        [System.ObsoleteAttribute("Use RunTargets() instead. Will be removed in version 2.0.0.")]
        public static void Run() { }
        [System.ObsoleteAttribute("Use RunTargets() instead. Will be removed in version 2.0.0.")]
        public static void Run(System.Collections.Generic.IEnumerable<string> args) { }
        [System.ObsoleteAttribute("Use RunTargetsAsync() instead. Will be removed in version 2.0.0.")]
        public static System.Threading.Tasks.Task RunAsync(System.Collections.Generic.IEnumerable<string> args) { }
        [System.ObsoleteAttribute("Use RunTargetsAsync() instead. Will be removed in version 2.0.0.")]
        public static System.Threading.Tasks.Task RunAsync() { }
        public static void RunTargets() { }
        public static void RunTargets(System.Collections.Generic.IEnumerable<string> args) { }
        public static System.Threading.Tasks.Task RunTargetsAsync(System.Collections.Generic.IEnumerable<string> args) { }
        public static System.Threading.Tasks.Task RunTargetsAsync() { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target(string name) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn) { }
        public static void Target(string name, System.Func<System.Threading.Tasks.Task> action) { }
        public static void Target(string name, System.Action action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Func<TInput, System.Threading.Tasks.Task> action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
        public static void Target(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Action action) { }
        public static void Target<TInput>(string name, System.Collections.Generic.IEnumerable<string> dependsOn, System.Collections.Generic.IEnumerable<TInput> forEach, System.Action<TInput> action) { }
    }
}